From 2ed421530522b2335bff96cc2abda9fd004144fa Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 17:22:55 +0000
Subject: [PATCH 5/5] context: Teach it all about automatic job counts

If we have 'auto' set in the eopkg.conf file, we'll default to finding
the maximum number of jobs available. This will ensure all users have
a sane build config set without having to edit eopkg.conf anymore.

This differs slightly to the current eopkg auto implementation of jobs+1
which is a bit taxing for most systems, so will be corrected.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index 7835f39..9b1fe14 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -16,6 +16,7 @@ from . import console_ui
 import pisi.config
 import os
 import shutil
+import multiprocessing
 
 # This speed flag set was originally from autospec in
 # Clear Linux Project For Intel Architecture.
@@ -151,7 +152,7 @@ class BuildConfig:
 
     ld_as_needed = True  # Make this configurable at some point.
 
-    jobcount = 2
+    jobcount = 4
 
     def get_flags(self, t):
         """ Simple switch to grab a set of flags by a type """
@@ -296,12 +297,18 @@ class YpkgContext:
         jobs = conf.values.build.jobs
         if "-j" in jobs:
             jobs = jobs.replace("-j", "")
+        elif jobs == "auto":
+            try:
+                jobs = multiprocessing.cpu_count()
+            except Exception as e:
+                console_ui.emit_warning(
+                    "BUILD", "Failed to detect CPU count, defaulting to 4")
         try:
             jcount = int(jobs)
             self.build.jobcount = jcount
         except Exception as e:
             console_ui.emit_warning("BUILD",
-                                    "Invalid job count of {}, defaulting to 2".
+                                    "Invalid job count of {}, defaulting to".
                                     format(jobs))
 
         self.global_archive_dir = conf.values.dirs.archives_dir
-- 
2.15.0

